datafusion-postgres/src/handlers.rs

```diff
@@ -509,19 +594,116 @@ impl ExtendedQueryHandler for DfSessionService {
 
         let (_, plan) = &portal.statement.statement;
 
-        let param_types = plan
-            .get_parameter_types()
-            .map_err(|e| PgWireError::ApiError(Box::new(e)))?;
+        // Enhanced parameter type inference with fallback for NULL + NULL scenarios
+        let param_types = match plan.get_parameter_types() {
+            Ok(types) => types,
+            Err(e) => {
+                // Check for specific planning errors related to NULL arithmetic operations
+                if matches!(e, datafusion::error::DataFusionError::Plan(_)) {
+                    let error_msg = e.to_string();
+                    if error_msg
+                        .contains("Cannot get result type for arithmetic operation Null + Null")
+                        || error_msg.contains("Invalid arithmetic operation: Null + Null")
```

Comment:
again, please don't judge error type with its message. this message is subject to change.